src/main/java/org/terasology/launcher/TerasologyLauncher.java

```diff
@@ -96,33 +102,91 @@ public void start(final Stage initialStage) {
         initProxy();
         initLanguage();
 
+        if (getParameters().getRaw().contains("--cleanup")) {
+            cleanupStart(initialStage);
+        } else {
+            normalStart(initialStage);
+        }
+    }
+
+    private void normalStart(Stage initialStage) {
         final Task<LauncherConfiguration> launcherInitTask = new LauncherInitTask(initialStage);
 
         showSplashStage(initialStage, launcherInitTask);
         Thread initThread = new Thread(launcherInitTask);
 
-        launcherInitTask.setOnSucceeded(new EventHandler<WorkerStateEvent>() {
-            @Override
-            public void handle(final WorkerStateEvent workerStateEvent) {
-                try {
-                    LauncherConfiguration config = launcherInitTask.getValue();
-                    if (config == null) {
-                        throw new LauncherStartFailedException("Launcher configuration was `null`.");
-                    } else if (config instanceof NullLauncherConfiguration) {
-                        logger.info("Closing the launcher ... (empty configuration, probably due to update)");
-                        Platform.exit();
-                    } else {
-                        showMainStage(config);
-                    }
-                } catch (IOException | LauncherStartFailedException e) {
-                    openCrashReporterAndExit(e);
+        launcherInitTask.setOnSucceeded((final WorkerStateEvent workerStateEvent) -> {
+            try {
+                LauncherConfiguration config = launcherInitTask.getValue();
+                if (config == null) {
+                    throw new LauncherStartFailedException("Launcher configuration was `null`.");
+                } else if (config instanceof NullLauncherConfiguration) {
+                    logger.info("Closing the launcher ... (empty configuration, probably due to update)");
+                    Platform.exit();
+                } else {
+                    showMainStage(config);
                 }
+            } catch (IOException | LauncherStartFailedException e) {
+                openCrashReporterAndExit(e);
             }
         });
 
         initThread.start();
     }
 
+    private void cleanupStart(Stage stage) {
+        logger.info("Starting cleanup tool because --cleanup argument was specified");
```

Comment:
Maybe just print "Starting in clean-up mode" to be more generic here. If you want to log the command line argument, do that when checking the parameters.